src/components/content/credential/AddCredential.tsx

```diff
@@ -0,0 +1,248 @@
+            dataIndex: 'description',
+        },
+    ];
+
+    const isContainsEmpty = (credentialVariableList: CredentialVariable[]) => {
+        if (credentialVariableList.length === 0) {
+            return false;
+        }
+        return credentialVariableList.some(
+            (credentialVariable) => credentialVariable.isMandatory && !credentialVariable.value
+        );
+    };
+
+    const submit = (credentialVariables: CredentialVariables) => {
+        if (!isContainsEmpty(credentialVariables.variables)) {
+            const createCredential: CreateCredential = credentialVariables as CreateCredential;
+            const userName: string | null = localStorage.getItem(usernameKey);
+            if (userName === null) {
+                return;
+            }
+            createCredential.xpanseUser = userName;
+            void CredentialsManagementService.addCredential(createCredential)
+                .then(() => {
+                    getCredentials();
+                    getTipInfo('success', 'Add Credential Success!');
+                    onReset();
+                    onCancel();
+                })
+                .catch(() => {
+                    getTipInfo('error', 'Add Credential Failed!');
+                });
+        }
+    };
+
+    const tailLayout = {
+        wrapperCol: { offset: 10, span: 16 },
+    };
+
+    const onReset = () => {
+        clear();
+        form.resetFields();
+    };
+
+    return (
+        <div className={'credential-from'}>
+            <Form
+                ref={formRef}
+                form={form}
+                labelCol={{ span: 4 }}
+                wrapperCol={{ span: 14 }}
+                layout='horizontal'
+                style={{ maxWidth: 1000 }}
+                onFinish={submit}
+            >
+                <div className={'credential-from-input'}>
+                    <Form.Item label='Name' name='name' rules={[{ required: true }]}>
+                        <Input />
+                    </Form.Item>
+
+                    <Form.Item label='Csp' name='csp' rules={[{ required: true, message: 'Please Select Csp!' }]}>
+                        <Select onSelect={handleCspSelect}>
+                            {Object.values(CredentialVariables.csp).map((csp: CredentialVariables.csp) => {
+                                return (
+                                    <Select.Option key={csp} value={csp}>
+                                        {csp}
+                                    </Select.Option>
+                                );
+                            })}
+                        </Select>
+                    </Form.Item>
+                    <Form.Item
+                        label='Type'
+                        name='type'
+                        rules={[{ required: true, message: 'Please Select The Type of Credential!' }]}
+                    >
+                        <Select onSelect={handleCredentialTypeSelect}>
```

Comment:
we still need a "loading"  to be displayed while the data is being loaded via API.